### ISSUE TITLE
init command throws JSON syntax error

### ISSUE THREAD

**Issue by kristiansaarela** [firebase/firebase-tools#4596]
### [REQUIRED] Environment info

**firebase-tools:** 11.0.1

**Platform:** Windows 10

### [REQUIRED] Test case

`firebase init database --debug` shouldn't throw an JSON syntax error.

### [REQUIRED] Steps to reproduce

Run `firebase init database --debug`
Choose "Use an existing project"
Choose any project
Use default file name for Realtime Database security rules

### [REQUIRED] Expected behavior

Database init finished successfully, not with JSON syntax error

### [REQUIRED] Actual behavior

? Are you ready to proceed? Yes

=== Project Setup

First, let's associate this project directory with a Firebase project. 
You can create multiple project aliases by running firebase use --add, 
but for now we'll just set up a default project.

? Please select an option: Use an existing project
[2022-05-30T12:30:16.316Z] >>> [apiv2][query] GET https://firebase.googleapis.com/v1beta1/projects pageSize=100
[2022-05-30T12:30:16.981Z] <<< [apiv2][status] GET https://firebase.googleapis.com/v1beta1/projects 200
[2022-05-30T12:30:16.982Z] <<< [apiv2][body] GET https://firebase.googleapis.com/v1beta1/projects [omitted]
? Select a default Firebase project for this directory: fir-kris-test (firebase-kris-test)
i  Using project fir-kris-test (firebase-kris-test)

=== Database Setup
i  database: ensuring required API firebasedatabase.googleapis.com is enabled...
[2022-05-30T12:30:33.150Z] >>> [apiv2][query] GET https://serviceusage.googleapis.com/v1/projects/fir-kris-test/services/firebasedatabase.googleapis.com [none]
[2022-05-30T12:30:34.083Z] <<< [apiv2][status] GET https://serviceusage.googleapis.com/v1/projects/fir-kris-test/services/firebasedatabase.googleapis.com 200
[2022-05-30T12:30:34.084Z] <<< [apiv2][body] GET https://serviceusage.googleapis.com/v1/projects/fir-kris-test/services/firebasedatabase.googleapis.com [omitted]
+  database: required API firebasedatabase.googleapis.com is enabled

[2022-05-30T12:30:34.090Z] >>> [apiv2][query] GET https://firebasedatabase.googleapis.com/v1beta/projects/fir-kris-test/locations/-/instances/fir-kris-test-default-rtdb [none]
[2022-05-30T12:30:34.487Z] <<< [apiv2][status] GET https://firebasedatabase.googleapis.com/v1beta/projects/fir-kris-test/locations/-/instances/fir-kris-test-default-rtdb 200
[2022-05-30T12:30:34.489Z] <<< [apiv2][body] GET https://firebasedatabase.googleapis.com/v1beta/projects/fir-kris-test/locations/-/instances/fir-kris-test-default-rtdb {"name":"projects/4564564564/locations/europe-west1/instances/fir-kris-test-default-rtdb","project":"projects/4564564654","databaseUrl":"https://fir-kris-test-default-rtdb.europe-west1.firebasedatabase.app","type":"DEFAULT_DATABASE","state":"ACTIVE"}

Firebase Realtime Database Security Rules allow you to define how your data should be
structured and when your data can be read from and written to.

? What file should be used for Realtime Database Security Rules? database.rules.json
[2022-05-30T12:31:02.589Z] >>> [apiv2][query] GET https://fir-kris-test-default-rtdb.europe-west1.firebasedatabase.app/.settings/rules.json [none][2022-05-30T12:31:02.772Z] <<< [apiv2][status] GET https://fir-kris-test-default-rtdb.europe-west1.firebasedatabase.app/.settings/rules.json 200
[2022-05-30T12:31:02.773Z] <<< [apiv2][body] GET https://fir-kris-test-default-rtdb.europe-west1.firebasedatabase.app/.settings/rules.json "{\n  /* Visit https://firebase.google.com/docs/database/security to learn more about security rules. */\n  \"rules\": {\n    \".read\": false,\n    \".write\": false\n  }\n}"

Error: Unable to parse JSON: SyntaxError: Unexpected token / in JSON at position 4

**Comment by macksal** [comment]
Does your rules file actually have comments in it? Comments are not part of the JSON standard so you should remove them.

**Comment by kristiansaarela** [comment]
I didn't create it. I'm trying to use `firebase init database ` command. Last line indicates that firebase is downloading `rules.json` file, which has comment in it and that breaks it.

**Comment by macksal** [comment]
Ah, it is downloading the rules that already exist in the project, and these include comments by default.

I'm able to replicate this by adding comments through the web interface. An easy workaround is just to remove the comment from the RTDB rules editor in the Firebase console (web).

It still probably represents an oversight on Google's part to allow comments to be added in the web interface, then expect the downloaded file to be valid JSON, so it likely deserves a fix.

**Comment by kristiansaarela** [comment]
I agree that some kind of improvements should be applied. Specially since it was 2nd command in Firebase documentation (`firebase init`), and it failing by default does not provide confidence.

I've removed comments from rules in Firebase console (web) and it works now. 

Thank you for helping.

**Comment by bkendall** [comment]
Huh, this is curious. I'll raise this with the rules team and see if the API is supposed to be able to parse out the comments, or if the CLI has to remove them before submitting them (which seems wrong).

**Comment by bkendall** [comment]
Ah, I see what's happening. When the new request library is making the request, it automatically tries to parse the response as JSON which is incorrect 'cause _this_ version of "JSON" can have comments. I'll adjust the requests so it just looks at them as strings and that should clear up the issue.
